File: app/src/main/java/com/hjq/demo/other/NetworkUsageLimiter.java
package com.hjq.demo.other;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUsageLimiter {

    private Context context;
    private long timeLimit; // 时间限制，例如30分钟

    public NetworkUsageLimiter(Context context, long timeLimit) {
        this.context = context;
        this.timeLimit = timeLimit;

        IntentFilter filter = new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION);
        context.registerReceiver(receiver, filter);
    }

    private BroadcastReceiver receiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            if (ConnectivityManager.CONNECTIVITY_ACTION.equals(intent.getAction())) {
                ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
                NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
                if (activeNetwork != null && activeNetwork.isConnected()) {
                    // 网络连接变为有效，开始计时
                    // TODO: 开始计时逻辑

                }
            }
        }
    };

    // 停止监听网络变化
    public void stopListening() {
        context.unregisterReceiver(receiver);
    }

}

File: app/src/main/java/com/hjq/demo/other/NetworkUsageLimiter.kt
package com.hjq.demo.other

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.ConnectivityManager
import android.net.NetworkInfo

class NetworkUsageLimiter(private val context: Context, private val timeLimit: Long) {

    private val receiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            if (ConnectivityManager.CONNECTIVITY_ACTION == intent.action) {
                val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
                val activeNetwork = cm.activeNetworkInfo
                if (activeNetwork != null && activeNetwork.isConnected) {
                    // 网络连接变为有效，开始计时
                    // TODO: 开始计时逻辑
                }
            }
        }
    }

    init {
        val filter = IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION)
        context.registerReceiver(receiver, filter)
    }

    fun stopListening() {
        context.unregisterReceiver(receiver)
    }
}
